File: evaluation/microservice-mongodb-morphia-java/src/main/java/cz/cuni/mff/java/microservice/model/embedded/CustomerEOnlyCNameCNation.java
package cz.cuni.mff.java.microservice.model.embedded;

import dev.morphia.annotations.Embedded;

@Embedded
public class CustomerEOnlyCNameCNation {
    private int c_custkey;
    private String c_name;
    private int c_nationkey;
    NationEOnlyNNameNRegion c_nation;

    public CustomerEOnlyCNameCNation() {}

    public CustomerEOnlyCNameCNation(int c_custkey, String c_name, int c_nationkey,
                                     NationEOnlyNNameNRegion c_nation) {
        this.c_custkey = c_custkey;
        this.c_name = c_name;
        this.c_nationkey = c_nationkey;
        this.c_nation = c_nation;
    }

    public int get_c_custkey() { return c_custkey; }

    public int get_c_nationkey() { return c_nationkey; }
}

File: evaluation/microservice-mongodb-mongodbentities-csharp/Model/Embedded/CustomerEOnlyCNameCNation.cs
namespace MongoDBEntitiesMicroservice.Model.Embedded 
{
    public class CustomerEOnlyCNameCNation
    {
        public int c_custkey { get; set; }
        public string c_name { get; set; }
        public int c_nationkey { get; set; }
        public NationEOnlyNNameNRegion c_nation { get; set; }

        public CustomerEOnlyCNameCNation() { }

        public CustomerEOnlyCNameCNation(int c_custkey, string c_name, int c_nationkey, NationEOnlyNNameNRegion c_nation)
        {
            this.c_custkey = c_custkey;
            this.c_name = c_name;
            this.c_nationkey = c_nationkey;
            this.c_nation = c_nation;
        }
    }
}
